nautobot/ui/src/components/core/Table.js

```diff
@@ -0,0 +1,48 @@
+import {
+    Checkbox,
+    Table,
+    Thead,
+    Tr,
+    Th,
+    Tbody,
+    Td,
+} from "@nautobot/nautobot-ui";
+
+import TableItem from "@components/core/TableItem";
+
+// A standard Nautobot table. This _may_ be beneficial to move into nautobot-ui
+export default function NautobotTable({ data, headers }) {
+    return (
+        <Table>
+            <Thead>
+                <Tr>
+                    <Th>
+                        <Checkbox></Checkbox>
+                    </Th>
+                    {headers.map(({ name, label }) => (
+                        <Th key={name}>{label}</Th>
+                    ))}
+                </Tr>
+            </Thead>
+            <Tbody>
+                {data.map((item) => (
+                    <Tr key={item.id}>
+                        <Td>
+                            <Checkbox></Checkbox>
+                        </Td>
+                        {headers.map((header, idx) => (
+                            <Td key={idx}>
+                                <TableItem
+                                    name={header.name}
+                                    obj={item[header.name]}
+                                    url={window.location.pathname + item["id"]}
+                                    link={idx === 0}
+                                />
+                            </Td>
+                        ))}
+                    </Tr>
+                ))}
+            </Tbody>
+        </Table>
+    );
+}
```

Comment:
this kind of table is a good use case for `TableRenderer` with [`enableMultiRowSelection: true`](https://tanstack.com/table/v8/docs/api/features/row-selection#enablemultirowselection) parameter. `TableRenderer` is basically an abstraction over `@tanstack/react-table` and exposes the majority of its APIs. in this case it would look something like:

```suggestion
import {
    createColumnHelper,
    TableRenderer,
    useTableRenderer,
} from "@nautobot/nautobot-ui";
import { useCallback, useMemo } from "react";

import TableItem from "@components/core/TableItem";

export default function NautobotTable({ data, headers }) {
    const columnHelper = useMemo(() => createColumnHelper(), []);

    const columns = useMemo(
        () =>
            headers.map(({ name, label }, idx) =>
                columnHelper.accessor(name, {
                    cell: ({
                        getValue,
                        row: {
                            original: { id },
                        },
                    }) => (
                        <TableItem
                            name={name}
                            obj={getValue()}
                            url={window.location.pathname + id}
                            link={idx === 0}
                        />
                    ),
                    header: label,
                })
            ),
        [columnHelper, headers]
    );

    const onRowSelectionChange = useCallback(() => {
        // Do something.
    }, []);

    const table = useTableRenderer({
        columns,
        data,
        enableMultiRowSelection: true,
        onRowSelectionChange,
    });

    return <TableRenderer table={table} />;
}

```

Reply:
https://github.com/nautobot/nautobot/pull/3531#discussion_r1160026432